webapp/src/components/call_widget/component.tsx

```diff
@@ -1440,7 +1440,12 @@ export default class CallWidget extends React.PureComponent<Props, State> {
                 body={body}
                 confirmText={confirmText}
                 declineText={isHost ? null : formatMessage({defaultMessage: 'Leave call'})}
-                onClose={() => this.setState({recDisclaimerDismissedAt: Date.now()})}
+                onClose={() => {
+                    this.setState({recDisclaimerDismissedAt: Date.now()});
+                    if (window.currentCallData) {
+                        window.currentCallData.recording.promptDismissedAt = Date.now();
+                    }
+                }}
```

Comment:
This is ugly, but since it's just this component there's no point in pulling it out and using a `useCallbasck`.